vunit/sim_if/vsim_simulator_mixin.py

```diff
@@ -312,10 +312,14 @@ def _run_persistent(self, common_file_name, load_only=False):
         except Process.NonZeroExitCode:
             return False
 
-    def simulate(self, output_path, test_suite_name, config, elaborate_only):
+    def simulate(self, output_path, test_suite_name, config, elaborate_only, precompiled=None):
         """
         Run a test bench
         """
+
+        with open(self._sim_cfg_file_name, "w") as ofile:
```

Comment:
What's the motivation for this change?

Reply:
This was a half-baked commit, for just presenting the general idea, in the end I want to support all the simulators.
Though ghdl is nice to start with. So as to delimit the commit.
That being said, by lint also complains about differing function signature and between the overridden base function and the subclass function, so all 'simulate' functions will need to be updated

Reply:
I updated the VUnit argument to only allow --precompiled for ghdl and riviera pro.
Riviera pro set experimental.